apps/condo/domains/property/components/panels/Builder/BuildingPanelView.tsx

```diff
@@ -68,21 +70,178 @@ const FLOOR_CONTAINER_STYLE: React.CSSProperties = { display: 'block' }
 const FULLSCREEN_HEADER_STYLE: React.CSSProperties = { marginBottom: '28px', alignItems: 'center' }
 const UNIT_TYPE_ROW_GUTTER: RowProps['gutter'] = [42, 0]
+const TOOLTIP_MAX_CONTACT_DETAILS = 5
+
+interface IUnitModalProps {
+    property: PropertyType
+    unit: BuildingUnit
+    contactsLoading: boolean
+    contacts: Array<ContactType>
+    contactsError: boolean
+}
+
+export const UnitModal: React.FC<IUnitModalProps> = ({ property, unit, contactsLoading, contacts }) => {
+    const intl = useIntl()
+    const FieldUnitNameMessage = intl.formatMessage({ id: 'field.Name' })
+    const FieldUnitTypeMessage = intl.formatMessage({ id: 'field.UnitType' })
+    const UnitTypeMessage = intl.formatMessage({ id: `field.UnitType.${unit.unitType}` }).toLowerCase()
+
+    const FieldContactNameMessage = intl.formatMessage({ id: 'field.FullName.short' })
+    const FieldPhoneMessage = intl.formatMessage({ id: 'Phone' })
+    const RolePhoneMessage = intl.formatMessage({ id: 'field.Role' })
+
+    const ContactsMessage = intl.formatMessage({ id: 'global.section.contacts' })
+    const GoToContactMessage = intl.formatMessage({ id: 'pages.condo.property.map.modal.goToContacts' })
+
+    const NoContactsTitleMessage = intl.formatMessage({ id: 'pages.condo.property.map.modal.noContacts.emptyView.title' })
+    const NoContactsSubtitleMessage = intl.formatMessage({ id: 'pages.condo.property.map.modal.noContacts.emptyView.subtitle' })
+    const NoContactsButtonLabel = intl.formatMessage({ id: 'AddContact' })
+
+    return (
+        <Row gutter={[0, 36]}>
+            <Col span={24}>
+                <List 
+                    dataSource={[
+                        {
+                            label: FieldUnitNameMessage,
+                            value: unit.label,
+                        },
+                        {
+                            label: FieldUnitTypeMessage,
+                            value: UnitTypeMessage,
+                        },
+                    ]}/>
+            </Col>
+            {(!contactsLoading && Array.isArray(contacts) && contacts.length > 0) && (
+                <Col span={24}>
+                    <Typography.Title level={4}>{ContactsMessage}</Typography.Title>
+                </Col>
+            )}
+            {(!contactsLoading && Array.isArray(contacts) && contacts.length === 0) && (
+                <BasicEmptyListView spaceSize={16} image='/dino/searching@2x.png'>
+                    <Space align='center' direction='vertical' size={8}>
+                        <Typography.Title level={3}>{NoContactsTitleMessage}</Typography.Title>
+                        <Typography.Text type='secondary'>{NoContactsSubtitleMessage}</Typography.Text>
+                    </Space>
+                    <Button type='primary' href={`contact/create?initialValues=${JSON.stringify( { property: property.id, unitName: unit.label, unitType: unit.unitType })}`}>{NoContactsButtonLabel}</Button>
```

Comment:
```suggestion
                    <Button type='primary' href={`/contact/create?initialValues=${JSON.stringify( { property: property.id, unitName: unit.label, unitType: unit.unitType })}`}>{NoContactsButtonLabel}</Button>
```

Reply:
Maybe redirect user back to map after creating a contact?

Reply:
I love this idea, let me do this :-)